libraries-server-2/src/main/java/com/baeldung/netty/customhandlersandlisteners/listener/ChannelInfoListener.java

```diff
@@ -0,0 +1,31 @@
+package com.baeldung.netty.customhandlersandlisteners.listener;
+
+import java.time.Instant;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.util.concurrent.GenericFutureListener;
+
+public class ChannelInfoListener implements GenericFutureListener<ChannelFuture> {
+
+    private final String event;
+
+    public ChannelInfoListener(String event) {
+        this.event = event;
+    }
+
+    @Override
+    public void operationComplete(ChannelFuture future) throws Exception {
+        Channel channel = future.channel();
+        String status = "OK";
+
+        if (!future.isSuccess()) {
+            status = "FAILED";
+            future.cause()
+                .printStackTrace();
+        }
+
+        System.out.println(String.format("%s - channel#%s %s: %s", Instant.now(), channel.id()
```

Comment:
We can replace println wih printf so we can get rid of String.format

Reply:
Suggestion included, draft updated.